### ISSUE TITLE
Build failure with Godot 3.2.3.stable.mono on Linux

### ISSUE THREAD

**Issue by akien-mga** [youkaicountry/yknytt#76]
I'm trying to build the current `master` branch (779d2a1), but I can't get the project to compile on my system with Godot 3.2.3.stable.mono.

System:
- Mageia Cauldron (Linux)
- Mono 6.12.0.90 from Mono RPM repos, MSBuild 16.6.0 for Mono
- Also tested: Dotnet SDK 3.1.403 for Microsoft RPM repos

When I open the project, I get those errors in the terminal:
```
ERROR: _create_autoload: Script does not inherit a Node: res://knytt/GDKnyttDataStore.cs.
   At: editor/editor_autoload_settings.cpp:357.
ERROR: _create_autoload: Script does not inherit a Node: res://knytt/GDKnyttKeys.cs.
   At: editor/editor_autoload_settings.cpp:357.
```
That's probably expected because those scripts haven't been compiled yet, so Godot can't load them as autoload.

Then when I try to build the solution, I get many errors:
```
Project "YKnytt.sln" (Restore target(s)):
	Message: Building solution configuration "Debug|Any CPU".
	Project "YKnytt.csproj" (_IsProjectRestoreSupported target(s)):
	Done building project "YKnytt.csproj".
	NuGetMessageTask: Determining projects to restore...
	Project "YKnytt.csproj" (_GenerateRestoreProjectPathWalk target(s)):
	Done building project "YKnytt.csproj".
	Project "YKnytt.csproj" (_IsProjectRestoreSupported target(s)):
	Done building project "YKnytt.csproj".
	Project "YKnytt.csproj" (_GenerateRestoreGraphProjectEntry target(s)):
	Done building project "YKnytt.csproj".
	Project "YKnytt.csproj" (_GenerateProjectRestoreGraph target(s)):
	Done building project "YKnytt.csproj".
	RestoreTask: Committing restore...
	RestoreTask: Assets file has not changed. Skipping assets file writing. Path: /home/akien/tmp/knytt/yknytt/.mono/temp/obj/project.assets.json
	RestoreTask: Restored /home/akien/tmp/knytt/yknytt/YKnytt.csproj (in 53 ms).
	RestoreTask: 
	RestoreTask: NuGet Config files used:
	RestoreTask:     /home/akien/.config/NuGet/NuGet.Config
	RestoreTask: 
	RestoreTask: Feeds used:
	RestoreTask:     https://api.nuget.org/v3/index.json
	RestoreTask: All projects are up-to-date for restore.
Done building project "YKnytt.sln".
Project "YKnytt.sln" (Build target(s)):
	Message: Building solution configuration "Debug|Any CPU".
	Project "YKnytt.csproj" (default targets):
		Skipping target "GenerateTargetFrameworkMonikerAttribute" because all output files are up-to-date with respect to the input files.
		Skipping target "CoreGenerateAssemblyInfo" because all output files are up-to-date with respect to the input files.
		Csc: /usr/lib/mono/msbuild/Current/bin/Roslyn/csc.exe /noconfig /unsafe- /checked- /nowarn:1701,1702,1701,1702 /nostdlib+ /errorreport:prompt /warn:4 /define:GODOT;GODOT_X11;GODOT_PC;TRACE;TOOLS;DEBUG;NETFRAMEWORK;NET472 /highentropyva+ /reference:/home/akien/tmp/knytt/yknytt/.mono/assemblies/Debug/GodotSharp.dll /reference:/home/akien/tmp/knytt/yknytt/.mono/assemblies/Debug/GodotSharpEditor.dll /reference:/usr/lib/mono/4.7.2-api/mscorlib.dll /reference:/usr/lib/mono/4.7.2-api/System.Core.dll /reference:/usr/lib/mono/4.7.2-api/System.Data.dll /reference:/usr/lib/mono/4.7.2-api/System.dll /reference:/usr/lib/mono/4.7.2-api/System.Drawing.dll /reference:/usr/lib/mono/4.7.2-api/System.IO.Compression.FileSystem.dll /reference:/usr/lib/mono/4.7.2-api/System.Numerics.dll /reference:/usr/lib/mono/4.7.2-api/System.Runtime.Serialization.dll /reference:/usr/lib/mono/4.7.2-api/System.Xml.dll /reference:/usr/lib/mono/4.7.2-api/System.Xml.Linq.dll /reference:/usr/lib/mono/4.7.2-api/Facades/netstandard.dll /debug+ /debug:portable /filealign:512 /optimize- /out:/home/akien/tmp/knytt/yknytt/.mono/temp/obj/Debug/YKnytt.dll /subsystemversion:6.00 /target:library /warnaserror- /utf8output /deterministic+ /langversion:7.3 knytt/GDAreaLayer.cs knytt/GDAreaTiles.cs knytt/GDKnyttArea.cs knytt/GDKnyttAssetManager.cs knytt/GDKnyttAudioChannel.cs knytt/GDKnyttBackground.cs knytt/GDKnyttCamera.cs knytt/GDKnyttDataStore.cs knytt/GDKnyttGame.cs knytt/GDKnyttKeys.cs knytt/GDKnyttSettings.cs knytt/GDKnyttWorld.cs knytt/GDKnyttWorldImpl.cs knytt/juni/ClimbCheckers.cs knytt/juni/CloudParticles.cs knytt/juni/DeathParticles.cs knytt/juni/DoubleJump.cs knytt/juni/GroundChecker.cs knytt/juni/Juni.cs knytt/juni/JuniAudio.cs knytt/juni/JuniMotionParticles.cs knytt/juni/JuniState.cs knytt/juni/Umbrella.cs knytt/objects/banks/bank0/NoClimb.cs knytt/objects/banks/bank0/NoJump.cs knytt/objects/banks/bank0/PowerItem.cs knytt/objects/banks/bank0/SavePoint.cs knytt/objects/banks/bank0/Shift.cs knytt/objects/banks/bank0/Sign.cs knytt/objects/banks/bank0/Warp.cs knytt/objects/banks/bank0/Win.cs knytt/objects/banks/bank1/LiquidPool.cs knytt/objects/banks/bank1/WaterBlock.cs knytt/objects/banks/bank1/Waterfall.cs knytt/objects/banks/bank10/BouncerEnemy.cs knytt/objects/banks/bank12/Ghost.cs knytt/objects/banks/bank12/GhostBlock.cs knytt/objects/banks/bank12/GhostMarker.cs knytt/objects/banks/bank12/GhostObject.cs knytt/objects/banks/bank12/NoWall.cs knytt/objects/banks/bank13/FlyBot.cs knytt/objects/banks/bank14/PassiveWalker.cs knytt/objects/banks/bank15/Door.cs knytt/objects/banks/bank15/ProximityBlock.cs knytt/objects/banks/bank15/Updraft.cs knytt/objects/banks/bank16/InvisibleBarrier.cs knytt/objects/banks/bank16/InvisibleBlock.cs knytt/objects/banks/bank16/Spring.cs knytt/objects/banks/bank2/BuzzFlyer.cs knytt/objects/banks/bank2/CircleBird.cs knytt/objects/banks/bank2/Elemental.cs knytt/objects/banks/bank2/FlySpike.cs knytt/objects/banks/bank3/Cat.cs knytt/objects/banks/bank4/ToastMonster.cs knytt/objects/banks/bank6/TrapCeiling.cs knytt/objects/banks/bank7/Cloud.cs knytt/objects/banks/bank7/Rain.cs knytt/objects/banks/bank7/Raindrop.cs knytt/objects/banks/bank7/RaindropObject.cs knytt/objects/banks/bank8/SimpleDecoration.cs knytt/objects/banks/bank8/Star.cs knytt/objects/banks/common/FollowerCreature.cs knytt/objects/CosineMod.cs knytt/objects/GDKnyttBaseObject.cs knytt/objects/GDKnyttObjectFactory.cs knytt/objects/GDKnyttObjectLayer.cs knytt/objects/GDObjectLayers.cs knytt/objects/IParticleFetch.cs knytt/RawAudioPlayer.cs knytt/RawAudioPlayer2D.cs knytt/ui/ClickPlayer.cs knytt/ui/Cutscene.cs knytt/ui/FadeLayer.cs knytt/ui/GameButton.cs knytt/ui/GameContainer.cs knytt/ui/InfoScreen.cs knytt/ui/info_panel/InfoPanel.cs knytt/ui/info_panel/ItemInfo.cs knytt/ui/InputOption.cs knytt/ui/InputScreen.cs knytt/ui/LevelSelection.cs knytt/ui/LocationLabel.cs knytt/ui/MainMenu.cs knytt/ui/particles/MenuCloud.cs knytt/ui/particles/ParticleCircle.cs knytt/ui/PausePanel.cs knytt/ui/SettingsScreen.cs knytt/ui/SlotButton.cs knytt/ui/touch/TouchPanel.cs knytt/ui/touch/TouchSettings.cs knytt/ui/touch/TouchSettingsScreen.cs knytt/ui/UICanvasLayer.cs Properties/AssemblyInfo.cs "/home/akien/tmp/knytt/yknytt/.mono/temp/obj/Debug/.NETFramework,Version=v4.7.2.AssemblyAttributes.cs" /home/akien/tmp/knytt/yknytt/.mono/temp/obj/Debug/YKnytt.AssemblyInfo.cs /warnaserror+:NU1605
		Csc: Using shared compilation with compiler from directory: /usr/lib/mono/msbuild/Current/bin/Roslyn
		Properties/AssemblyInfo.cs(19,28): error CS8357: The specified version string contains wildcards, which are not compatible with determinism. Either remove wildcards from the version string, or disable determinism for this compilation [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		.mono/temp/obj/Debug/YKnytt.AssemblyInfo.cs(14,12): error CS0579: Duplicate 'System.Reflection.AssemblyCompanyAttribute' attribute [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		.mono/temp/obj/Debug/YKnytt.AssemblyInfo.cs(15,12): error CS0579: Duplicate 'System.Reflection.AssemblyConfigurationAttribute' attribute [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		.mono/temp/obj/Debug/YKnytt.AssemblyInfo.cs(18,12): error CS0579: Duplicate 'System.Reflection.AssemblyProductAttribute' attribute [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		.mono/temp/obj/Debug/YKnytt.AssemblyInfo.cs(19,12): error CS0579: Duplicate 'System.Reflection.AssemblyTitleAttribute' attribute [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		.mono/temp/obj/Debug/YKnytt.AssemblyInfo.cs(20,12): error CS0579: Duplicate 'System.Reflection.AssemblyVersionAttribute' attribute [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/GDAreaTiles.cs(3,7): error CS0246: The type or namespace name 'YKnyttLib' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/GDKnyttArea.cs(2,7): error CS0246: The type or namespace name 'YKnyttLib' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/GDKnyttAssetManager.cs(4,7): error CS0246: The type or namespace name 'YKnyttLib' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/GDKnyttAssetManager.cs(5,7): error CS0246: The type or namespace name 'YUtil' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/GDKnyttBackground.cs(2,7): error CS0246: The type or namespace name 'YKnyttLib' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/GDKnyttCamera.cs(2,7): error CS0246: The type or namespace name 'YUtil' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/GDKnyttGame.cs(2,7): error CS0246: The type or namespace name 'YKnyttLib' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/GDKnyttKeys.cs(2,7): error CS0246: The type or namespace name 'IniParser' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/GDKnyttKeys.cs(3,7): error CS0246: The type or namespace name 'IniParser' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/GDKnyttSettings.cs(2,7): error CS0246: The type or namespace name 'IniParser' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/GDKnyttSettings.cs(3,7): error CS0246: The type or namespace name 'IniParser' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/GDKnyttWorld.cs(2,7): error CS0246: The type or namespace name 'YKnyttLib' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/GDKnyttWorld.cs(3,7): error CS0246: The type or namespace name 'YKnyttLib' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/GDKnyttWorldImpl.cs(3,7): error CS0246: The type or namespace name 'YKnyttLib' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/juni/Juni.cs(3,7): error CS0246: The type or namespace name 'YKnyttLib' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/juni/Juni.cs(4,7): error CS0246: The type or namespace name 'YUtil' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/juni/Juni.cs(5,7): error CS0246: The type or namespace name 'YUtil' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/juni/Juni.cs(6,14): error CS0246: The type or namespace name 'YKnyttLib' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/objects/banks/bank0/Shift.cs(3,7): error CS0246: The type or namespace name 'YKnyttLib' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/objects/banks/bank0/Shift.cs(4,14): error CS0246: The type or namespace name 'YKnyttLib' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/objects/banks/bank1/LiquidPool.cs(3,7): error CS0246: The type or namespace name 'YUtil' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/objects/banks/bank1/Waterfall.cs(2,7): error CS0246: The type or namespace name 'YUtil' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/objects/banks/bank12/Ghost.cs(2,7): error CS0246: The type or namespace name 'YUtil' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/objects/banks/bank14/PassiveWalker.cs(2,7): error CS0246: The type or namespace name 'YUtil' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/objects/banks/bank2/BuzzFlyer.cs(3,7): error CS0246: The type or namespace name 'YUtil' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/objects/banks/bank2/CircleBird.cs(3,7): error CS0246: The type or namespace name 'YUtil' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/objects/banks/bank2/FlySpike.cs(2,7): error CS0246: The type or namespace name 'YUtil' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/objects/banks/bank3/Cat.cs(2,7): error CS0246: The type or namespace name 'YUtil' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/objects/banks/bank4/ToastMonster.cs(2,7): error CS0246: The type or namespace name 'YUtil' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/objects/banks/bank7/Rain.cs(3,7): error CS0246: The type or namespace name 'YKnyttLib' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/objects/banks/bank7/RaindropObject.cs(2,7): error CS0246: The type or namespace name 'YKnyttLib' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/objects/banks/bank8/Star.cs(2,7): error CS0246: The type or namespace name 'YUtil' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/objects/GDKnyttBaseObject.cs(2,7): error CS0246: The type or namespace name 'YKnyttLib' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/objects/GDKnyttObjectFactory.cs(3,7): error CS0246: The type or namespace name 'YKnyttLib' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/objects/GDKnyttObjectLayer.cs(2,7): error CS0246: The type or namespace name 'YKnyttLib' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/objects/GDObjectLayers.cs(3,7): error CS0246: The type or namespace name 'YKnyttLib' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/ui/InfoScreen.cs(2,7): error CS0246: The type or namespace name 'YKnyttLib' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/ui/info_panel/ItemInfo.cs(2,7): error CS0246: The type or namespace name 'YKnyttLib' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/ui/LevelSelection.cs(6,7): error CS0246: The type or namespace name 'YKnyttLib' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/ui/LocationLabel.cs(2,7): error CS0246: The type or namespace name 'YKnyttLib' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/ui/MainMenu.cs(3,7): error CS0246: The type or namespace name 'YKnyttLib' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/ui/UICanvasLayer.cs(2,14): error CS0246: The type or namespace name 'YKnyttLib' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/GDKnyttWorldImpl.cs(5,33): error CS0246: The type or namespace name 'KnyttWorld' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/GDKnyttWorldImpl.cs(9,31): error CS0115: 'GDKnyttWorldImpl.bytesToSound(byte[])': no suitable method found to override [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/GDKnyttWorldImpl.cs(14,31): error CS0115: 'GDKnyttWorldImpl.bytesToTexture(byte[])': no suitable method found to override [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/GDKnyttWorldImpl.cs(19,29): error CS0115: 'GDKnyttWorldImpl.externalFileExists(string)': no suitable method found to override [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/GDKnyttWorldImpl.cs(26,31): error CS0115: 'GDKnyttWorldImpl.getExternalSound(string)': no suitable method found to override [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/GDKnyttWorldImpl.cs(34,31): error CS0115: 'GDKnyttWorldImpl.getExternalTexture(string)': no suitable method found to override [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/GDKnyttWorldImpl.cs(42,38): error CS0115: 'GDKnyttWorldImpl.getExternalWorldData(string)': no suitable method found to override [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/GDKnyttWorldImpl.cs(50,31): error CS0115: 'GDKnyttWorldImpl.getSystemSound(string)': no suitable method found to override [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/GDKnyttWorldImpl.cs(55,31): error CS0115: 'GDKnyttWorldImpl.getSystemTexture(string)': no suitable method found to override [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/GDKnyttWorldImpl.cs(60,38): error CS0115: 'GDKnyttWorldImpl.getSystemWorldData(string)': no suitable method found to override [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/ui/info_panel/ItemInfo.cs(15,28): error CS0246: The type or namespace name 'JuniValues' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/GDKnyttArea.cs(9,12): error CS0246: The type or namespace name 'KnyttArea' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/GDKnyttArea.cs(35,36): error CS0246: The type or namespace name 'KnyttPoint' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/GDKnyttArea.cs(42,12): error CS0246: The type or namespace name 'KnyttPoint' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/GDKnyttArea.cs(56,46): error CS0246: The type or namespace name 'KnyttArea' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/objects/banks/bank0/Shift.cs(9,5): error CS0246: The type or namespace name 'KnyttShift' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/ui/UICanvasLayer.cs(50,29): error CS0246: The type or namespace name 'PowerNames' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/juni/Juni.cs(40,9): error CS0246: The type or namespace name 'KnyttPoint' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/juni/Juni.cs(42,9): error CS0246: The type or namespace name 'JuniValues' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/ui/LevelSelection.cs(10,5): error CS0246: The type or namespace name 'KnyttWorldManager<>' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/GDKnyttAssetManager.cs(56,39): error CS0246: The type or namespace name 'KnyttPoint' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/GDKnyttAssetManager.cs(57,27): error CS0246: The type or namespace name 'KnyttPoint' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/ui/LevelSelection.cs(175,13): error CS0246: The type or namespace name 'KnyttWorldManager<>' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/ui/LevelSelection.cs(187,13): error CS0246: The type or namespace name 'KnyttWorldManager<>' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/GDKnyttGame.cs(111,23): error CS0246: The type or namespace name 'KnyttPoint' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/GDKnyttGame.cs(111,40): error CS0246: The type or namespace name 'KnyttPoint' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/GDKnyttAssetManager.cs(17,2): error CS0246: The type or namespace name 'ObjectCache<,>' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/juni/Juni.cs(175,23): error CS0246: The type or namespace name 'PowerNames' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/GDKnyttAssetManager.cs(18,2): error CS0246: The type or namespace name 'ObjectCache<,>' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/GDKnyttAssetManager.cs(19,2): error CS0246: The type or namespace name 'ObjectCache<,>' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/GDKnyttAssetManager.cs(20,2): error CS0246: The type or namespace name 'ObjectCache<,>' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/ui/LevelSelection.cs(25,21): error CS0246: The type or namespace name 'KnyttWorldManager<>' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/GDKnyttGame.cs(184,33): error CS0246: The type or namespace name 'KnyttPoint' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/GDKnyttAssetManager.cs(21,2): error CS0246: The type or namespace name 'ObjectCache<,>' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/GDKnyttAssetManager.cs(21,14): error CS0246: The type or namespace name 'KnyttPoint' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/GDKnyttGame.cs(190,25): error CS0246: The type or namespace name 'KnyttPoint' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/juni/Juni.cs(355,47): error CS0246: The type or namespace name 'KnyttPoint' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/objects/GDKnyttBaseObject.cs(8,12): error CS0246: The type or namespace name 'KnyttPoint' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/objects/GDKnyttBaseObject.cs(9,12): error CS0246: The type or namespace name 'KnyttPoint' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/objects/GDKnyttBaseObject.cs(27,28): error CS0246: The type or namespace name 'KnyttPoint' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/objects/GDKnyttBaseObject.cs(27,76): error CS0246: The type or namespace name 'KnyttPoint' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/ui/LocationLabel.cs(15,32): error CS0246: The type or namespace name 'KnyttPoint' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/objects/GDKnyttObjectFactory.cs(105,56): error CS0246: The type or namespace name 'KnyttPoint' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/objects/GDKnyttObjectFactory.cs(7,31): error CS0246: The type or namespace name 'KnyttPoint' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/objects/GDKnyttObjectFactory.cs(132,64): error CS0246: The type or namespace name 'KnyttPoint' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/objects/GDKnyttObjectFactory.cs(121,12): error CS0246: The type or namespace name 'KnyttPoint' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/GDKnyttKeys.cs(12,9): error CS0246: The type or namespace name 'IniData' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/objects/GDKnyttObjectFactory.cs(125,32): error CS0246: The type or namespace name 'KnyttPoint' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/GDKnyttSettings.cs(8,19): error CS0246: The type or namespace name 'IniData' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/objects/GDObjectLayers.cs(10,17): error CS0246: The type or namespace name 'KnyttPoint' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/objects/banks/bank15/Door.cs(9,16): error CS0246: The type or namespace name 'YKnyttLib' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/objects/banks/bank15/Door.cs(11,38): error CS0246: The type or namespace name 'YKnyttLib' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/GDKnyttWorld.cs(11,12): error CS0246: The type or namespace name 'KnyttRectPaging<>' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/GDKnyttWorld.cs(54,32): error CS0246: The type or namespace name 'KnyttPoint' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/GDKnyttWorld.cs(59,19): error CS0246: The type or namespace name 'KnyttPoint' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/GDKnyttWorld.cs(65,40): error CS0246: The type or namespace name 'KnyttPoint' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/GDKnyttWorld.cs(77,29): error CS0246: The type or namespace name 'KnyttPoint' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
		knytt/objects/GDKnyttObjectLayer.cs(14,27): error CS0246: The type or namespace name 'KnyttPoint' could not be found (are you missing a using directive or an assembly reference?) [/home/akien/tmp/knytt/yknytt/YKnytt.csproj]
	Done building project "YKnytt.csproj" -- FAILED.
Done building project "YKnytt.sln" -- FAILED.
```

I tried both options in Editor Settings > Mono > Builds > Build Tool (i.e. "dotnet CLI" and "MSBuild (Mono)" on Linux) while removing `res://.mono` before each attempt, but it fails the same with both tools.

**Comment by youkaicountry** [comment]
Interesting, I have been making all builds on a Windows machine. At first glance it looks like it is somehow having trouble with the DLLs I am including. I'll try to build it on my Linux box and see if I can get to the bottom of it.

**Comment by akien-mga** [comment]
I found the issue, the project is named "YKnytt" in `project.godot` so Godot wants `YKnytt.{csproj,sln}`, but yours are named `Yknytt.{csproj,sln}` (lowercase `k`). So Godot would create new, empty `YKnytt.{csproj,sln}` which fail building.

Case matters for cross-platform compatibility :)

**Comment by youkaicountry** [comment]
Hah, it certainly does :)

I usually dev on Linux... That's what I get for developing this project on a Windows machine!

Thanks for figuring it out, I'll get a fix pushed in a bit

**Comment by youkaicountry** [comment]
Oh nevermind, you have a PR already :)

**Comment by youkaicountry** [comment]
Resolved by PR #77
